Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM ubuntu:22.04
+
+RUN apt-get update && apt-get install -y \
+                    git \
+                    wget \
+                    flex \
+                    bison \
+                    gperf \
+                    python3 \
+                    python3-pip \
+                    python3-venv \
+                    python3-setuptools \
+                    cmake \
+                    ccache \
+                    libffi-dev \
+                    libssl-dev \
+                    dfu-util \
+                    libusb-1.0-0 \
+                    libgcrypt-dev \
+                    libglib2.0-dev \
+                    libfdt-dev \
+                    libpixman-1-dev \
+                    zlib1g-dev \
+                    ninja-build \
+                    libslirp-dev
+
+
+
+RUN mkdir -p ~/esp && cd ~/esp && git clone -b v4.4.7 --recursive https://github.com/espressif/esp-idf.git && cd esp-idf && ./install.sh && /bin/bash -c "source  ./export.sh"
+
+RUN echo 'source /root/esp/esp-idf/export.sh' >> $HOME/.bashrc
+
+RUN mkdir -p ~/qemu && cd ~/qemu && git clone --recursive https://github.com/espressif/qemu.git && cd qemu \
```

Comment:
Same here as [this](https://github.com/XmasHackathon2023/esp_wireguard/pull/3/files#r1760973590)